recipe/build.sh

```diff
@@ -13,17 +13,26 @@ test -n "${AUTORECONF}" \
 # need macosx-version-min flags set in cflags and not cppflags
 export CFLAGS="$CFLAGS $CPPFLAGS"
 
+if [[ "$target_platform" == "osx-"* ]]; then
+    USESSL="--with-openssl=${PREFIX} --with-secure-transport --with-default-ssl-backend=openssl"
```

Comment:
I didn't read the docs, so sorry in advance. Why --with-default-ssl-backend=openssl` if we build with secure transport?

Reply:
My (limited) understanding is you can use either transport, but I will try to educate myself further.

Reply:
From: https://everything.curl.dev/usingcurl/tls/backends.html

```
When curl is built with multiple TLS backends, it can be told which one to use each time it is started. It is always built to use a specific one by default unless one is asked for.

If you invoke curl --version for a curl with multiple backends it mentions MultiSSL as a feature in the last line. The first line includes all the supported TLS backends with the non-default ones within parentheses.

To set a specific one to get used, set the environment variable CURL_SSL_BACKEND to its name.
```

curl --version outputs:

```
% curl --version
curl 8.11.1 (arm64-apple-darwin20.0.0) libcurl/8.11.1 OpenSSL/3.0.15 (SecureTransport) zlib/1.2.13 zstd/1.5.6 libssh2/1.11.1 nghttp2/1.57.0
Release-Date: 2024-12-11
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```

There is `MultiSSL` in the second line and `OpenSSL/3.0.15 (SecureTransport) ` in the first. So OpenSSL is the default, and `SecureTransport` can be used by setting `CURL_SSL_BACKEND`